packages/@aws-cdk/aws-lambda/lib/languages/lambda-builder.ts

```diff
@@ -0,0 +1,106 @@
+import assets = require('@aws-cdk/assets');
+import { BuildAsset } from '@aws-cdk/assets';
+import cdk = require('@aws-cdk/cdk');
+import { join } from 'path';
+import { Code } from '../code';
+import { CfnFunction } from '../lambda.generated';
+import { Runtime } from '../runtime';
+
+interface LambdaBuilderRequest {
+  jsonrpc: string;
+  method: string;
+  id: number;
+  params: {
+    __protocol_version: string;
+    capability: {
+      language: string;
+      dependency_manager: string;
+      application_framework: string | null;
+    },
+    source_dir: string;
+    artifacts_dir: string;
+    scratch_dir: string;
+    manifest_path: string;
+    runtime: string;
+    optimizations: object;
+    options: object;
+  }
+}
+
+export interface LambdaBuilderCodeProps {
+  path: string;
+  runtime: Runtime;
+  language: string;
+  dependencyManager: string;
+  manifestName: string;
+}
+
+export class LambdaBuilderCode extends Code {
+  public readonly isInline: boolean = false;
+
+  protected readonly path: string;
+  protected readonly runtime: Runtime;
+
+  protected readonly language: string;
+  protected readonly dependencyManager: string;
+  protected readonly manifestName: string;
+
+  protected asset?: BuildAsset;
+
+  constructor(props: LambdaBuilderCodeProps) {
+    super();
+    this.path = props.path;
+    this.runtime = props.runtime;
+    this.language = props.language;
+    this.dependencyManager = props.dependencyManager;
+    this.manifestName = props.manifestName;
+  }
+
+  public bind(construct: cdk.Construct) {
+    // If the same AssetCode is used multiple times, retain only the first instantiation.
```

Comment:
Not super familiar with the inner workings of CDK so excuse my naive-ness here. 

If you are building many functions, won't this place everything into the same `artifacts_dir` without any way to untangle the directory? I am not sure if you want this retained over different builds. `aws-lambda-builders` will place the artifact into into `artifacts_dir`. This is the isolation model for a function. So if you are building two Go functions, the `artifacts_dir` will have two binaries. Instead, you will want to create sub directories in some model CDK understands. This will allow you to map back artifacts to functions without clobbering them all together.